src/Tasks/ResolveSDKReference.cs

```diff
@@ -1251,7 +1251,7 @@ private void CreateResolvedReferenceItem(string targetConfiguration, string targ
                     AddResolutionWarning("ResolveSDKReference.MaxPlatformVersionNotSpecified", projectName, DisplayName, Version, targetPlatformIdentifier, targetPlatformVersionFromItem.ToString(), targetPlatformIdentifier, targetPlatformVersion.ToString());
                 }
 
-                if (!String.IsNullOrEmpty(TargetPlatform) && !String.Equals(targetPlatformIdentifier, TargetPlatform))
+                if (!String.IsNullOrEmpty(TargetPlatform) && !String.Equals(targetPlatformIdentifier, TargetPlatform) && (!String.Equals("UAP", TargetPlatform, StringComparison.OrdinalIgnoreCase) || !String.Equals(targetPlatformIdentifier, "windows", StringComparison.OrdinalIgnoreCase)))
```

Comment:
I'd create a hashset of `supportedAliases` or something akin to that. Then we can comment next to the variable a link to the issue, and this new bit of code will be self-documenting.

Reply:
Actually, a dictionary <string,string> might make more sense.  `if (string.Equals(TargetPlatform, supportedAliases[TargetPlatform])`, this way we can directly map windows to UAP.

My priority in this case would be clarity + potential future cases similar to this, though admittedly I'm not 100% sure how often it would happen so I could be over-engineering.

Either way we should link back to this PR/issue.

Reply:
I would consider making a local function so you can be a bit more verbose.

Reply:
I made a dictionary. I don't think it's quite as nice as just doing the string comparisons for just one alias, but if we have multiple, it starts to be clearer.